.gitpod.yml

```diff
@@ -1,28 +1,54 @@
 image:
   file: .gitpod.dockerfile
 tasks:
-  - init: npm_config_yes=true pnpm setup:project
+  - init: npm_config_yes=true pnpm setup:project && gp sync-done setup
     command: mkdir -p /workspace/data && mongod --dbpath /workspace/data 
-  - command: redis-server
+  - init: gp sync-await setup
+    command: redis-server
   - name: Shared Library
+    init: gp sync-await setup
     command: cd libs/shared && pnpm start
   - name: DAL
+    init: gp sync-await setup
     command: cd libs/dal && pnpm start
   - name: TESTING LIB
+    init: gp sync-await setup
     command: cd libs/testing && pnpm start
   - name: API
+    init: gp sync-await setup
     command: cd apps/api && pnpm start
   - name: WS
+    init: gp sync-await setup
     command: cd apps/ws && pnpm start
   - name: WEB
-    command: cd apps/web && pnpm start
+    init: gp sync-await setup
+    command: |
+      cd apps/web
+      API_URL=`gp url 3000`
+      sed -i "/REACT_APP_API_URL=/c\REACT_APP_API_URL=$API_URL" .env
+      pnpm envsetup && pnpm start
+  - name: WIDGET
+    init: gp sync-await setup
+    command: |
+      cd apps/widget
+      API_URL=`gp url 3000`
+      WS_URL=`gp url 3002`
+      sed -i "/REACT_APP_API_URL=/c\REACT_APP_API_URL=$API_URL" .env
+      sed -i "/REACT_APP_WS_URL=/c\REACT_APP_WS_URL=$WS_URL" .env
+      pnpm envsetup && pnpm start:dev
```

Comment:
```suggestion
      pnpm envsetup && pnpm start
```

Reply:
It's default port is 3000, which is used by API. If we don't set to other port, there will be collision.

Reply:
Then it feels right to update the Widget in the script also to provide a different port.

Reply:
like this?
```suggestion
      PORT=4500
      pnpm envsetup && pnpm start
```